bindgen/private/bindgen.bzl

```diff
@@ -64,27 +65,109 @@ def rust_bindgen_library(
         name = name + "__bindgen",
         header = header,
         cc_lib = cc_lib,
         bindgen_flags = bindgen_flags or [],
         clang_flags = clang_flags or [],
-        tags = ["manual"],
+        tags = sub_tags,
+        **bindgen_kwargs
+    )
+
+    compile_data = kwargs.pop("compile_data", []) + [name + "__bindgen.compile_data"]
+    native.filegroup(
+        name = name + "__bindgen.compile_data",
+        srcs = [name + "__bindgen"],
+        output_group = "bindgen_compile_data",
+        tags = sub_tags,
     )
 
+    for custom_tag in ["__bindgen", "no-clippy", "no-rustfmt"]:
+        tags = tags + ([] if custom_tag in tags else [custom_tag])
+
     rust_library(
         name = name,
         srcs = [name + "__bindgen.rs"],
-        tags = tags + ["__bindgen", "noclippy"],
-        deps = deps + [cc_lib],
+        deps = deps + [name + "__bindgen"],
+        compile_data = compile_data,
+        tags = tags,
         **kwargs
     )
 
+def _generate_cc_link_build_info(ctx, cc_lib):
+    """Produce the eqivilant cargo_build_script providers for use in linking the library.
+
+    Args:
+        ctx (ctx): The rule's context object
+        cc_lib (Target): The `rust_bindgen.cc_lib` target.
+
+    Returns:
+        The `BuildInfo` provider.
+    """
+    compile_data = []
+    linker_flags = []
+    linker_search_paths = []
+
+    for linker_input in cc_lib[CcInfo].linking_context.linker_inputs.to_list():
+        for lib in linker_input.libraries:
+            if lib.static_library:
+                linker_flags.append("-lstatic={}".format(lib.static_library.owner.name))
+                linker_search_paths.append(lib.static_library.dirname)
+                compile_data.append(lib.static_library)
+            elif lib.pic_static_library:
+                linker_flags.append("-lstatic={}".format(lib.pic_static_library.owner.name))
+                linker_search_paths.append(lib.pic_static_library.dirname)
+                compile_data.append(lib.pic_static_library)
+
+        linker_flags.extend(linker_input.user_link_flags)
+
+    if not compile_data:
+        fail("No static libraries found in {}".format(
+            cc_lib.label,
+        ))
+
+    empty_file = ctx.actions.declare_file("{}.empty".format(ctx.label.name))
+    ctx.actions.write(
+        output = empty_file,
+        content = "",
+    )
+
+    link_flags = ctx.actions.declare_file("{}.link_flags".format(ctx.label.name))
+    ctx.actions.write(
+        output = link_flags,
+        content = "\n".join(linker_flags),
+    )
+
+    link_search_paths = ctx.actions.declare_file("{}.link_search_paths".format(ctx.label.name))
+    ctx.actions.write(
+        output = link_search_paths,
+        content = "\n".join([
+            "-Lnative=${{pwd}}/{}".format(path)
+            for path in depset(linker_search_paths).to_list()
+        ]),
+    )
+
+    return BuildInfo(
+        out_dir = ctx.file._bindgen_out_dir,
+        rustc_env = empty_file,
```

Comment:
I'd be tempted to make theses optional rather than having to pass an empty file (and make an entire rule to make an empty dir), but I don't feel strongly

Reply:
Good idea, Done!